src/mlpack/methods/mean_shift/mean_shift_impl.hpp

```diff
@@ -216,8 +217,8 @@ inline void MeanShift<UseKernel, KernelType, MatType>::Cluster(
   {
     // Initial centroid is the seed itself.
     allCentroids.col(i) = pSeeds->unsafe_col(i);
-    for (size_t completedIterations = 0; completedIterations < maxIterations;
-         completedIterations++)
+    for (size_t completedIterations = 0; completedIterations < maxIterations
+      || forceConvergence; completedIterations++)
```

Comment:
Ah, I guess maybe I should have clarified a bit.  I thought that a `forceConvergence` parameter would run the mean shift algorithm for at least the minimum number of iterations necessary to produce one cluster.  So we could also add some part to the end of the inner loop like

```
if (forceConvergence && centroids.n_cols >= 1)
  break;
```

What do you think?

Reply:
So if `forceConvergence` is passed, we should only force the 1st seed to converge but not the others? 
From what I understand from the code, the mean shift algorithm runs for each seed centroid generated earlier up to the given `maxIterations` or it converges, so if `forceConvergence` is given I think we should be forcing all these seed centroids to converge.

Reply:
Ok, I can agree with that.  Let's leave it the way it is now then.